\chapter{Future work}
\chaptertoc{}
\paragraph{}At the interface between developmental biology and computer science, I address together questions about embryonic morphogenesis, computer vision, data science and data visualisation.
By developing novel and innovative computational algorithms I sought to quantify, model and better understand embryonic development at the single cell scale in whole organisms throughout development.

\paragraph{}More specifically, I want to understand how the different forms and functions of tissues, organs and organisms are obtained as a result of cell changes at the morphological and molecular level.
% I will look at embryogenesis at the single cell scale by computationally building and combining statistical averages of the development across two modalities: fluorescence microscopy images and spatial omics.
% To quantitatively understand how cells arrange in tissues, tissues in organs and organs in organisms, I will computationally build and combine statistical averages of embryonic development across two modalities: fluorescence microscopy images and spatial omics.
I will do so across three main axes/question:
\begin{enumerate}
	\item Morphological scale: \underline{How do cells organise in coherent structures?}
	\item Molecular scale: \underline{What genes control that organisation?}
	\item \emph{Morpholecular} scale: \underline{How do genes and cell organisation interplay?}
\end{enumerate}


% The first axis will be about looking at \emph{how do cells position and shape tissues} by finding novel ways to address \emph{how to build average representatives of developing organisms from 3D fluorescence microscopy time-series}.
% % To computer vision, machine learning and network analysis algorithms to reconstruct and analyse average representatives of developing organisms from 3D fluorescence microscopy time-series.

% The second axis will be about finding \emph{which are the genes responsible for major morphological events} by exploring \emph{how to build, store in memory and represent such large and complex 3D spatial omics dataset}.

% Finally, the third axis will be about understanding \emph{how cell molecular content and cell morphogenesis influence each other} by addressing \emph{how to integrate both morphological and molecular dataset into comprehensive average atlases of embryogenesis across multiple species}.

% Being at the interface of computer science and developmental biology, my research is and will be pursued in collaboration with novel and already established collaborations with developmental biologists and physicists.
\paragraph{}\textbf{Major expected outcomes of my future work:}\\
My future work will produce the following major outcomes:
\noindent\begin{enumerate}
	\item \underline{Developmental Biology}: {Quantitative and statistical understanding of cell dynamics during embryogenesis across multiple model and non-model organisms}
	\item \underline{Computer science}: {Improved algorithms to detect and track cells from fluorescence microscopy and algorithms to reconstruct and mine near single-cell spatial transcriptomics}
	\item \underline{Data science}: {Ways to display, browse and query datasets in a seamless manner}
	\item \underline{Dataset}: {Statistical atlases of the development of multiple model and non-model organisms}
\end{enumerate}
\section{Scientific context}
\paragraph{}A single fertilised oocyte generates a highly organised living organism composed of thousands to billions of cells.
Through divisions, shape changes, growth, rearrangement or programmed death, cells gradually form coherent groups that organise in tissues, themselves organising into organs and eventually creating a fully functional organism.
This process, called embryogenesis, is highly complex and requires precise spatial and temporal coordination with precise regulation of hundreds to millions of cells, their geometry, their spatial organisation and their molecular content, to eventually produce the final functional organism.
Despite its complexity, the result of this process must be reproducible: a fertilised oocyte must produce a functional adult.
% \paragraph{}Added to the complexity mentioned above, embryonic development
\paragraph{}One of the first steps to understanding the mechanisms driving any given process is to observe it as precisely as possible and quantify it.
Developmental biologists have been observing processes for centuries to understand the underlying rules governing the creation of an organism from a single fertilised egg.
\paragraph{}In the past ten years, technological breakthroughs have pushed the observations of biological samples to scales that were not previously accessible.
More specifically, the continuous development of light-sheet fluorescence microscopy (method of the year 2014~\cite{Meth:2015aa}) has enabled the observation of whole embryos, throughout their development, at the single cell resolution, for periods lasting from hours to days.
Another set of revolutionary methods are single-cell sequencing methods (methods of the year 2013~\cite{Meth:2014aa}, breakthrough of the year 2018~\cite{Crespi2018}) and more specifically spatially resolved single cell sequencing methods (method of the year 2020~\cite{Meth2021}) that allow to precisely measure the molecular content of all the cells in a given embryo, at a given time in the development.
\paragraph{}Altogether, these two complementary new methods have brought the realm of observation of the state of an organism to a whole new level.
But, while these novel methods allow highly resolved and sometimes prolonged observations of the organisms, they generate datasets that are too large and too complex to be analysed manually, and often even to be analysed using standard computational methods.
For example, light-sheet microscopes generate three-dimensional movies that weigh up to a few terabytes in memory and can contain tens of thousands of cells that must be detected and followed to quantify their dynamics.
Developing computational methods to perform such tasks automatically is currently a hot research area.
\paragraph{}Furthermore, single-cell sequencing methods measure the concentration of tens of thousands of different molecules in tens of thousands of cells generating datasets with dimensionality far higher than what has already been treated in biology.
Again, more and more groups are working on developing methods to treat these datasets.
\paragraph{}These new methods allow biologists to quantitatively peek at developmental processes at unprecedented scales and precision.
But, because of the extreme characteristics of these newly acquired datasets, new computational and mathematical methods are required to accurately extract the quantitative information buried within them.
Moreover, even once the measurements are extracted, it is often nontrivial to make sense of them nor to extract rules from these observations.
For that, too, novel theoretical methods have to be designed.
\paragraph{}This is why it is now often necessary to first answer computational questions to then answer biological ones.
\section{Objectives}
\paragraph{}My research programme consists in better understanding the interplay between cell dynamics and cell gene expression patterns.
I will do so by developing novel methods to extract biologically relevant quantitative information from the previously mentioned acquisition methods.
Another goal of my research program is to make this extracted quantitative information accessible to visualise, browse and query, hence easily available to me and the biology community.
This work will be carried in close relationship with other biologists, computer scientists and physicists while following these three axes:
\begin{itemize}
   \item \textbf{Axis 1}: \underline{Morphological}: Deciphering cell dynamics by quantify them in developing organisms from \emph{in-toto} recording and build dynamic averages of these quantifications
   \item \textbf{Axis 2}: \underline{Molecular}: Systematically identifying genes of interest by quantifying single cell molecular content of developing organisms from near single-cell spatial omics and build averages of these quantification
   \item \textbf{Axis 3}: \underline{Morpholecular}: Bridge morphological and molecular information to understand how they interplay to control embryogenesis at the single-cell scale
\end{itemize}
\begin{figure}[h!]
   \begin{center}
       \includegraphics[width=.8\textwidth]{../Figures/Figures_Figure 1.png}
       \caption{\textbf{Schematics of my research proposal} A. Recording of the development from microscopes or from spatial-single cell technics.
       Done by collaborators.
       B. Reconstruction of the acquired microscope time-series to quantify cell morphodynamics.
       C. Aggregation of the multiple morphodynamic reconstructions into an average atlas.
       D. Reconstruction of the spatial-omics datasets.
       E. Aggregation of the multiple spatial-omics reconstructions into an average atlas.
       F. Aggregation of the two atlases with different modalities into a multimodal single-cell atlas: The morpholecular atlas}
   \end{center}
\end{figure}
\paragraph{}My research programme aims at bringing novel insights on both model and non-model organisms in developmental biology.
% Led by my own biological questions and together with my current and future collaborators, the computational approaches that I will develop will always motivated by biology.
While the algorithms and libraries will be primarily developed to answer my own biological questions, they will most likely be useful to the larger biology community when they will want to observe their organisms quantitatively, at the single-cell scale.
Finally, the outputs of my research will be potentially useful to the applied physicist community wanting to develop data driven models of embryonic development.
\paragraph{}\underline{Host institution}: I have recently moved my own group to the Institut de Biologie du Développement de Marseille (IBDM, UMR 7288) in Marseille.
Therefore, the IBDM would be my preferred host institution.
The IBDM focuses on developmental biology which fits the biological questions I want to address.
Moreover, the IBDM already has interdisciplinary groups: a bio-computational group (B. Habermann, bioinformatics) and a biophysics group (P.-F. Lenne group, physical approaches to morphogenesis).
% I am the only image analysis group, even though many groups work with fluorescence imaging.
The IBDM greatly benefits from my current group as we are doing image analysis, spatial transcriptomics and data analysis as well as I am greatly benefiting from all the potential collaborations with groups generating dataset that would need quantitative analysis.
Moreover, my research about embryonic development across 6 different species fits well within specific groups such as the ones of P.-F. Lenne or A. Le Bivic, two groups with which I already have ongoing collaborations.
Other groups, such as the one of T. Lecuit, B. Prud’homme and V. Bertrand would be fruitful terrains for new collaborations.
\paragraph{}The IBDM is also part of the Turing Centre for living systems (CENTURI), a centre for interdisciplinary research within which I am currently a group leader.
% \paragraph{}Another choice for developing my research would be the La\-bo\-ra\-toire d’In\-for\-ma\-ti\-que et Sys\-tèmes (LIS, UMR 7020) also on the Aix-Marseille university campus.
% The LIS is currently hosting me and my group, and is a computer science laboratory where my research on image analysis and applied graph theory would fit well.
% I am currently already integrated, as a CENTURI group leader, within the group ACRO (algorithms, combinatorial and research operational) in the Calculus department where my research on applying graph theory to biological models fits one of the missions of the group, that is the application of their developed methods to real world problems.


\section{Morphodynamic atlases of embryonic development}
\paragraph{}To understand morphogenesis and embryogenesis, biologists observe the normal development of their organisms of choice~\cite{Wallingford2019}.
They also perform targeted disruptions on their organisms of interest and compare the disrupted development to the original one.
Both methods of investigation require looking at a set of samples, as precisely as possible.
Moreover, to compare two sets of behaviours (for example wild-type and perturbed) it is important to have a statistical view of the set of samples that were recorded to precisely measure significant differences.
In this axis I propose to develop computational methods to quantitatively look at and measure organisms in 3 dimensions, through time and at the single-cell scale.
I propose to also develop methods to aggregate multiple samples onto an average representative, a statistical atlas.
These methods extract the quantitative information from recordings of embryonic development made with fluorescence microscopes where cell membranes and/or cell nuclei are labelled~\cite{Huisken2004,Keller2008,Krzic2012,Medeiros2015}.
\paragraph{}To extract quantitative measurements from such movies and enable the quantification of organism development at the single cell scale it is necessary to develop methods to detect the cells, reconstruct their shape and track them throughout the recordings, preferably in a fully automatic fashion.
Once reconstructed, these movies can then be combined to build an average representative of the wild-type (normal) development.
To do so, the cell trajectories from different recordings of the organism, at the same developmental time points, must be combined to implement a more general statistical view of the average development.
These statistical views can then be used for a precise, quantitative description of the variability of wild-type behaviour~\cite{Wolff:2018aa,Li2019,Guignard2020}.
The statistical views can also be used to perform quantitative and statistically relevant comparisons between wild-type behaviour and perturbed one.
\paragraph{}Therefore, this axis aims at developing computational frameworks to reconstruct, align and aggregate recordings of cell morphodynamics into average statistical representatives (Fig.~\ref{fig:registration}).
The building of these atlases not only relies on the existence of recordings of the development but also on the method used will change with the properties of the imaged sample.
This axis is split into three sub-axes, each tackling the reconstruction and aggregation of different organisms that exhibit different degrees of variability and come from recordings made from either fluorescence light-sheet microscopy or bi-photon microscopy.
First, I will work with already acquired in-toto recordings of \textit{Parhyale hawaiensis}.
Second, I will work with 50, already acquired, \emph{in-toto} recordings of \textit{Drosophila melanogaster} that last 7 hours from the beginning of the gastrulation.
And third, I will work with already acquired recordings of cell aggregates, namely gastruloids, that mimic one of the most important developmental processes, gastrulation, where a group of undifferentiated cells gives rise to the three germ layers.
Each of these axes is supported by an independent biological question but a common computational one: \textbf{How to build a quantitative and statistical view of a developing biological system?}
\begin{figure}[!ht]
   \begin{center}
       \includegraphics[width=.95\textwidth]{../Figures/Figures_Figure 2.png}
       \caption{\textbf{Principles of time series registration.}
       A. A set of \(n\) time series of recorded embryos. The black ticks show the acquisition timepoints.
       The grey arrows show a mapping between homologous timepoints.
       B. The time series are registered in time.
       Potential time dilation and compression can be seen.
       C. All embryos, at all shared registered timepoints, are spatially registered onto the reference embryo (here Embryo 1).
       D. The transformations that allow to register the embryos onto the average space are computed using the pre-computed spatial transformations.
       E. Once all embryos are registered onto the average space, they can be averaged}\label{fig:registration}
   \end{center}
\end{figure}
\subsection{Morphological atlas of \textit{P. hawaiensis}}
\paragraph{}One of the most pervasive, yet poorly understood, hallmarks of most metazoans is bilateral symmetry~\cite{Genikhovich2017}.
The external left and right sides of bilaterians develop separately but somehow manage to produce symmetric matching halves, presumably via deterministic developmental programmes, homeostatic buffering mechanisms or combinations thereof.
These mechanisms are poorly understood~\cite{Wolpert2010,Grimes2019}.
In collaboration with Anastasios Pavlopoulos, a developmental biologist at the Institute of Molecular Biology and Biotechnology of the Foundation for Research and Technology Hellas in Heraklion, Greece, I will probe these mechanisms in the embryonic epithelial monolayer of the crustacean \textit{Parhyale hawaiensis} that acquires a stereotypic, geometrically ordered and bilaterally symmetric tissue architecture composed of less than a thousand cells, while maintaining the capacity to restore left/right symmetry after unilateral cell ablations.
Through live imaging with multi-view light-sheet fluorescence microscopy~\cite{Wolff:2018aa} (Fig.~\ref{fig:images}) and image analysis of intact and ablated embryos, \textbf{I will combine novel computer vision and data analysis algorithms to decompose the shaping of each side into the individual cell contributions.} {By doing so, I will quantify their levels of stereotypy and variability across sides, embryos and conditions during the normal emergence of bilateral symmetry and during its restoration after cell ablation}.
\paragraph{}In this project, I will advance the development of computational algorithms for aligning and integrating segmented image data (consisting of cell proliferation and cell death, cell shape changes and rearrangements) with cellular accuracy across sides, biological replicates and conditions.
I will also develop the mathematical framework for the unbiased quantification of left/right (a)symmetries across spatiotemporal scales based on the cellular observables collected.
To quantify the left/right (a)symmetries I will extend the algorithm that I used in my previous work on ascidians~\cite{Guignard2020}.
This algorithm~\cite{Zhang1996}, whose principles are similar to that of DNA sequence distance/alignment algorithms such as BLAST~\cite{Altschul1990}, compares lineage trees by defining the difference between two lineage trees \(T_1\) and \(T_2\) (let it be \(d(T_1, T_2)\)) as the minimum number of weighted operations that rewrites the binary tree \(T1\) into \(T2\): \(d(T_1, T_2)=\argmin_{d\in D}\sum_{o\in d} w_o.o\) where \(D\) is the set of all possible rewritings, \(d\) is a given rewriting, \(o\) is an atomic rewriting operation of \(d\) and \(w_o\) is the associated cost of the operation \(o\).
These measurements in wild-type and perturbed conditions will allow the detection of statistically relevant deviation from the wild-type behaviour in the perturbed embryos, pointing to the timing of interest during bilateral formation of \textit{P. hawaiensis} embryos.
These times will be used for single cell molecular content analysis (Axis 2).
\subsection{Morphological at\-las of \textit{Dro\-so\-phi\-la me\-la\-no\-gas\-ter}}
\paragraph{}\textit{Drosophila melanogaster} is one of the main model organisms for developmental biology, yet, to the best of my knowledge, there is no available extensive reconstruction of the cell dynamics of this fly.
A reconstruction of the average cell dynamics of the embryogenesis of \textit{D. melanogaster} would be a rich and precious asset to the community.
To remedy this missing information, \textbf{I will develop novel computational methods to build a 3D+time atlas of \textit{D. melanogaster} development}.
To do so, I will use 50 already existing 3D wild-type time-series provided by W. Lemon from the group of P.J. Keller at Janelia Research Campus, USA\@. These movies cover the embryonic development of the fruit fly \textit{D. melanogaster} for 7 hours on average, starting at the beginning of gastrulation with a frame rate of one 3D acquisition every 30 seconds.
The movies were acquired using multiview fluorescence light-sheet microscopy~\cite{Royer2016}.
Taking advantage of state-of-the-art cell detection and tracking algorithms that I have and helped developed (TGMM~\cite{Amat:2014aa} and SVF~\cite{McDole:2018aa}), I will first track cell dynamics across the entire time-series (Fig.~\ref{fig:images}).
I will then redefine the TARDIS framework I developed for building the mouse post-implantation atlas~\cite{McDole:2018aa} to scale to a high number of recorded embryos.
Currently, two key limitations prevent TARDIS from being scalable to constructing an atlas that integrates a cohort of embryos: the average embryo is biassed towards a reference embryo that must be arbitrarily chosen, and manual landmarks are required.
The manual process of landmark annotation in the current scheme prohibits scaling the method to a larger number of embryos.
I will remove the manual landmarks by automatically detecting clusters of cells that have specific, unique properties such as local density or velocity.
These clusters of cells will be used in place of the landmarks.
For example, the germband cells during its elongation and retraction can inform about the dorso-ventral and anterio-posterior orientation of the embryo and its developmental rate.
Based on these automatic landmarks, I will then use combinatorial optimisation algorithms to find the set of transformations that registers every embryo onto the space of the average embryo.
This set of transformations will be optimised to minimise the sum of the distances between the average and all the original embryos.
This set of transformations will create a genuinely unbiased average space for the average synthetic embryo, removing the potential bias towards a reference.
\subsection{Reconstruction of gastruloid development}
\paragraph{}As stated, morphogenesis is orchestrated by complex mechanisms involving biochemical signalling and mechanical interactions.
One remarkable event during the development of a given organism is the breaking of the initial symmetry.
This breaking is necessary to define the anterior and posterior parts of the embryo and later on, to define all the essential tissues.
Even though microscopy has dramatically improved within the past few years, recording symmetry breaking events at the single cell scale remains difficult in-vivo.
\begin{figure}[!th]
   \begin{center}
       \includegraphics[width=.95\textwidth]{../Figures/Figures_Figure 3.png}
       \caption{\textbf{Recordings and reconstructions.}
       Top left: 3D projections of a \textit{P. hawaïensis} embryo where the membranes and nuclei are labelled with fluorophores with two different excitation and emission wavelengths.
       Top row: 10h after egg laying (AEL), bottom row, 64h AEL.
       Top right: 3D projections of the reconstruction of 4 \textit{D. melanogaster} embryos during germband elongation.
       Bottom: 5 different gastruloids where the nuclei are labelled with the mesoderm marker Brachyury and the dividing cells are labelled with the mitotic marker Ph3.}\label{fig:images}
   \end{center}
\end{figure}
\paragraph{}For this reason, recent in-vitro systems mimicking mammalian embryogenesis have been developed.
It has been shown that when cultured under biomimetic conditions, embryonic stem cells aggregate and organise into structures that recapitulate the early developmental stages of mouse embryogenesis~\cite{vandenBrink2014}.
These systems of stem cell aggregates are called gastruloids and constitute convenient models for understanding how stem cells can self-organise to develop tissues.
They present another advantage: they allow high-throughput experiments, which is essential to study the variability from one sample to another and therefore the mechanisms ensuring the robustness of development.
In collaboration with S. Tlili in the group of P.-F. Lenne, at the IBDM, I am co-mentoring A. Gros, a Ph.D. student, to develop algorithms to reconstruct, track cells and quantify the observable developmental variability in gastruloids.
More specifically, in this project, the biological question addressed focuses on \textbf{the role of cell divisions during the morphogenesis of gastruloids, specifically during the formation of the anterio-posterior axis}.
The 3D geometry and complex dynamics of such multicellular systems require developing or adapting specific tools for imaging, segmentation and analysis.
To study the mechano-genetics coupling during symmetry breaking, I am currently and will continue to co-lead the development of a pipeline to quantify in 3D the heterogeneities in gene expression and mechanical properties in gastruloids.
This pipeline includes the production of experimental data, here immunostainings on fixed samples to image gene expression patterns with a two-photon microscope.
From the 3D images, the pipeline is an adaptation of a deep neural network~\cite{Weigert2020} to segment nuclei shapes, and extract a range of nuclear properties (volumes, deformations, divisions..), averaged over variable spatial scales and compared with genetic observables to look for spatial correlations with genetic observables, for instance, the mesoderm marker Brachyury.
Ultimately, I would like to probe the spatio-temporal scales of variability that could be observed during gastruloid development to enable the construction of a morphogenetic atlas of their development.
It would create an invaluable source of information for the rapidly growing field of organoids.
\section{sc3D: a fast and easy dive into spatial near single-cell omics}
\paragraph{}The first axis is focused on reconstructing cell morphodynamics and building statistical representatives of embryogenesis at the single cell scale.
While understanding cell movements and shape changes is important to understand embryogenesis, one crucial piece of information is missing from these measurements: cell molecular content.
Indeed, precise spatiotemporal orchestration of gene expression is required for proper embryonic development.
Recently, the development of high throughput single-cell omics (scO) technologies has generated comprehensive transcriptomic definitions of cell states within the embryo, shedding light on many mechanisms.
But, the design of the first scO methods is so that spatial information is lost (Fig.~\ref{fig:sc3D}).
In other words, while it is possible to measure the molecular content of every single cell within a given sample, it is not possible to retain their positional information.
Even more recently, efforts have been made to alleviate this issue.
Novel protocols, namely near single-cell spatial omics (scSO), have been newly developed to keep the spatial localization of the measured cells (Fig.~\ref{fig:sc3D}).
As scO methods did when they came out, scSO methods bring new exciting computational challenges.
Amongst these are two unavoidable ones:
\begin{itemize}
   \item \textbf{How to incorporate spatial dependency information to the already complex high dimensional analysis?}
   \item \textbf{How to visualise and browse efficiently such large datasets?}
\end{itemize}
In this axis, I propose to develop two computational libraries, namely sc3D and sc3D-viewer, to efficiently handle and analyse (sc3D) and dynamically display and browse (sc3D-viewer) scSO datasets.
These two libraries split the axis in two sub-axes:
\begin{itemize}
   \item \textbf{Sub-axis 2.1: sc3D}.
   The sc3D library will extend the already existing AnnData~\cite{Wolf2018} library, allowing handling of classic single-cell transcriptomic datasets.
   sc3D will include the organisation of the spatial (position and neighbourhood relationship) information in efficient data structures.
   sc3D will also contain scSO specific preprocessing and analysis methods.
   \item \textbf{Sub-axis 2.2: sc3D-viewer}.
   sc3D-viewer will be based on the sc3D data structure and will allow the display of scSO datasets in 3D.
   sc3D-viewer will allow the projection, on the fly, of expression and/or co-expression of measured genes or any post-computed metrics.
\end{itemize}
\begin{figure}[!th]
   \begin{center}
       \includegraphics[width=.8\textwidth]{../Figures/Figures_Figure 4.png}
       \caption{\textbf{Single cell omics principles.}
       A. In classic single cell omics, the cells are dissociated from each other before being recorded.
       This process clears away the spatial information.
       As a result, the processed data are often projected in 2 dimensions, in a way that groups cells together according to their molecular content.
       B. In spatial single cell omics (here I am referring specifically to SlideseqV2~\cite{Stickels2021}), 2 dimensional sections are made from a 3D sample.
       These sections are then deposited onto an array of barcoded beads of the size of a single cell.
       Through gravity, each cell of the 2D slice falls onto a bead and is barcoded according to that bead.
       This primary barcode allows us to record the 2D position of the cell and therefore to replace its measurement in space a-posteriori.}\label{fig:sc3D}
   \end{center}
\end{figure}
\subsection{sc3D}
\paragraph{}sc3D will address one of the primary challenges brought by scSO: how to store relevant spatial information efficiently.
As a first step and the way it is usually done, spatial information is stored as if it were a classic measurement: three values representing the x, y, z coordinates of the position of the measured beads.
While memory efficient, it does not directly encompass the neighbouring information between beads and therefore prevents the use of many computational methods designed for spatial datasets.
To resolve this problem, I propose to extend the state-of-the-art data structure for Python (AnnData~\cite{Wolf2018}) to add a network representation of the beads where the nodes of the networks are the beads and the edges of the network are the neighbourhood relationships between beads.
The neighbourhood relationship will be computed as the Gabriel graph of the spatialised beads similarly to~\cite{McDole:2018aa} where it was proven to be a representative neighbourhood reconstruction while keeping a low computational time complexity.
\paragraph{}Another challenge that has arisen for scSO methods is the reconstruction, in 3D, of the datasets.
Indeed, scSO datasets are usually acquired as a set of 2 dimensional stacks.
However, the spatial coherence is often not kept from slice to slice.
Therefore algorithms such as Tangram~\cite{Biancalani2021}, STUtility~\cite{Bergenstrahle2020} or more recently PASTE~\cite{Zeira2022} have been developed.
Here I propose to incorporate the previously developed algorithms to sc3D so that users can choose amongst all the different methods.
I will also develop a new algorithm based on pre-computed cluster alignments.
Cell clusters can be computed using classic methods implemented in the reference libraries for scO: Seurat~\cite{Hao2021} or scanpy~\cite{Wolf2018} for example.
When correctly parameterized, these clusters represent the different tissues or sub-tissues.
I will take advantage of the fact that most tissues are continuous in space and use that property to align the different slices.
By going from the single cell/bead scale to the tissue scale, our algorithm will be significantly faster than the already existing algorithms.
\paragraph{}Because of the spatial nature of the datasets, some already existing scO methods need to be adapted and many new methods can be developed.
I will translate the already existing scO methods to incorporate the spatial component.
One example of such methods is detecting spatially differentially expressed genes ((S)DEG).
I will develop novel approaches to integrate spatial information into the detection of DEG in space.
Our SDEG detection method will be based on creating subnetworks of expressing cells and on the property that quasi-linearly links the density of a network and a random subnetwork of that network.
On top of our new method to detect SDEG, I will also incorporate state of the art methods such as Splotch~\cite{Aijo2019} or C-SIDE~\cite{Cable2021}.
\paragraph{}Finally, I will develop a new set of methods to build statistical atlases of scSO from a set of samples.
Thanks to the spatial nature of the dataset, I will be able to take advantage of my expertise in point cloud registration to develop methods to register scSO datasets against each other and therefore allow their aggregation into an average representation, building the first single cell omics 3D statistical atlases.
This average single-cell spatial atlas of the species or tissue recorded will enable the statistical comparison between wild-type/healthy samples and perturbed/sick samples allowing us to statistically measure the gene expression deviation of an abnormal sample to a normal one.
\subsection{sc3D-viewer}
\paragraph{}Extracting biologically relevant information from such complex and high dimensional datasets is challenging.
Exploring the datasets is the first step to orienting questions and helping experimentalists design experiments.
Such exploration has to be easy and seamless.
\paragraph{}To that aim, I will develop sc3D-viewer: a 3D viewer for scSO based on napari, a software initially dedicated to 3D visualisation of microscopy images.
sc3D-viewer will be both a way of exploring scSO datasets but also a graphical interface for sc3D to allow users to run the tools and methods proposed in sc3D without having to write code.
\paragraph{}Our viewer will aim not only to display the dataset in 3D but also to be intuitively interactive.
Among others, the viewer will allow the user to choose tissues or clusters to display or hide.
It will also allow the user to choose to project on the cells any measured gene expression or metrics such as the pseudo-time for example.
The user will also be able to have a dynamic interaction between the 3D view of their datasets and the more common 2D projections such as UMAPs and tSNEs representations.
sc3D-viewer will allow the user to select cells from any 2D projection and display them on the fly in the 3D viewer.
I will also enable the user to display multiple genes simultaneously to explore gene co-expression.
Finally, I will develop methods to compute meshes from any set of points to visualise the different tissues or classes better.
\paragraph{}As I have already developed a prototype: \href{https://www.github.com/guignardlab/napari-sc3D-viewer}{napari-sc3D-viewer}, I know that it will be possible to indeed have the rapid, seamless and in depth user experience that is required to explore such dense and large datasets.
\subsection{Usefulness to the biological community in general}
\paragraph{}I have already started to develop these tools in collaboration with the groups of F. Chen at the Broad Institute, Cambridge, MA, USA and A. Meissner at the Max Planck Institute for Molecular Genetics, Berlin, Germany.
The group of A. Meissner provided the samples, the group of F. Chen developed the method to measure the cell molecular content while keeping the spatial information and I developed the computational tools to reconstruct, analyse and visualise the generated datasets (see the document about the previous works for more details)~\cite{Sampath-Kuma:2022aa}.
\begin{figure}[b!]
   \begin{center}
       \includegraphics[width=.8\textwidth]{../Figures/Figures_Figure 5.png}
       \caption{\textbf{Single-cell spatial transcriptomics of an E8.5 mouse embryo.}
       A. Prototype of the 3D viewer for spatial single-cell datasets.
       Brachyury is projected on the cells.
       B. Example of an interaction where the user selected cells in the 2D projection on the left and the same cells are displayed on the 3D viewer.}\label{fig:sc3D-viewer}
   \end{center}
\end{figure}
\paragraph{}For this project, I built the first 3D molecular atlas of a mouse embryo at 8.5 and 9.0 days after fertilisation (Fig.~\ref{fig:sc3D-viewer}).
The atlas is a powerful tool for biologists who can perform in-silico 3D virtual in-situs for 27000 genes and further look in depth at tissues and genes of interest to quantify expression.
Furthermore, biologists can systematically look for which genes are locally expressed in their tissues of interest.
Currently developed for mouse data, expanding these methods to other organisms will be a valuable asset  to an even larger community.
\paragraph{}Finally, the tools developed for the mouse datasets from A. Meissner's group will be extended to the datasets that will later be acquired by the group of A. Pavlopoulos with whom I have an ongoing collaboration.
\section{Morpholecular atlases}
\paragraph{}\textbf{Axes 1} and \textbf{2} will produce extremely rich single-cell scale datasets of two different modalities: morphological and molecular.
By themselves, these datasets have led to significant biological discoveries, however, they are not sufficient yet to reflect the complex synergy between morphology and molecular.
In order to understand, at the single-cell scale, the interplays between the molecular content of a cell, group of cells or tissues and their morphodynamics, the next logical step is to integrate these two modalities together, morphological and molecular, into a common atlas.
The goal of this third axis is to \textbf{create the very first single cell atlas of cell morphodynamics together with their molecular content: a morpholecular atlas}.
\paragraph{}The morpholecular atlas, together with the adequate visualisation tools will provide biologists with an integrated view of the development of their favourite organisms.
If they are interested in the specific cell movement of a given tissue, they will be able systematically look for genes that are specifically expressed at the spatio-temporal position of their interest.
If they are interested in a specific gene, they will be able to systematically look at which morphological events are correlated with the specific expression of these genes.
\paragraph{}Building such morpholecular atlases comes with challenges, the main ones are likely the following:
\begin{enumerate}
   \item The two modalities will probably come from different samples, therefore, \textbf{how to map two samples, from different modalities, onto each other?}
   \item The recording of the single cell molecular modality is lethal to the sample, so it is only possible to record fixed snapshots, therefore, \textbf{how to propagate the static molecular information in time?}
   \item The construction of a morpholecular atlas will generate one of a kind data\-sets, \textbf{how to validate the quality of the atlas?}
   \item Such morpholecular atlases are extremely rich and span many dimensions, therefore, \textbf{how to easily and seamlessly browse and query such complex datasets?}
\end{enumerate}
\paragraph{}To answer these questions I will develop my first prototype of the morpholecular atlas on the epidermis of the \textit{Parhyale hawaïensis} embryo working in collaboration with A. Pavlopoulos who will provide the samples.
\textit{P. hawaïensis} is the ideal model for developing such a prototype since its morphological complexity is in between the complex one of the mouse and the simpler ones of highly stereotyped species such as \textit{C. elegans}.
Moreover, while it is still an emergent model system, many tools have been developed for genetic manipulation and imaging of the embryos.
\paragraph{}Such tools already exist and will allow the recordings of \textit{P. hawaïensis} to be done for both modalities in a single sample.
The embryos will be imaged using a light-sheet fluorescence microscope, then, at the end of the recording of the cell morphodynamics, the sample will be frozen and the spatial single cell RNA sequencing will be made with SlideSeqV2~\cite{Stickels2021}.
In doing so, I will be able to overlay both datasets from the same embryo at the same developmental time point.
\paragraph{}This approach of combining imaging and SlideSeqV2~\cite{Stickels2021} will be performed up to different timepoints in order to acquire the co-recordings of the spatial single cell transcriptomics and the cell dynamics at different developmental times.
These multimodal recordings will form a ground truth against which I will quantify the quality of the reconstruction of my methods.
Using these recordings will also allow the validation of the methods to propagate the molecular information in space.
\paragraph{}Using these morpholecular atlases, we will address the question of the bilateral symmetry of \textit{P. hawaïensis}.
It is known that \textit{P. hawaïensis} can recover their bilateral symmetry after specific drastic cell ablations, however, the mechanisms driving this recovery are not yet known.
By comparing the wild-type morpholecular atlases to the measurements of ablated embryos, we will find what genes are responsible for the recovery of the bilateral symmetry.
\paragraph{}Finally, I will extend the viewer developed for the Axis 2 in order to integrate spatio-temporal information and images.
To do so, I will take advantage of the fact that the software basis of the sc3D-viewer is napari, a software that was initially designed for 3D+time image visualisation.
